hello.go

```diff
@@ -31,6 +37,63 @@ type Data struct {
 	Region   string
 }
 
+func handleReceivedEvent(ctx context.Context, event cloudevents.Event) {
+	type LoggedEvent struct {
+		Severity  string            `json:"severity"`
+		EventType string            `json:"eventType"`
+		Message   string            `json:"message"`
+		Event     cloudevents.Event `json:"event"`
+	}
+	type PubSubMessage struct {
+		Data string `json:"data"`
+	}
+	type PubSubEvent struct {
+		Message PubSubMessage `json:"message"`
+	}
+
+	dataMessage := event.Data()
+
+	// In case of PubSub event, decode its payload to be printed in the message as-is.
+	if event.Type() == "google.cloud.pubsub.topic.v1.messagePublished" {
+		obj := &PubSubEvent{}
+		if err := event.DataAs(obj); err != nil {
+			fmt.Printf("Unable to decode PubSub data: %s\n", err.Error())
```

Comment:
you should also probably return something (a status code + message) to the client for debugging?

Reply:
you can use log.Printf and skip doing \n.

Reply:
Using log.Printf, either with `\n` or without, does not allow Cloud Logging to parse the output as JSON:

![image](https://user-images.githubusercontent.com/4989256/109201390-75d57080-77aa-11eb-9c0d-5fb77847ac3f.png)

As for returning the status code, it's not necessary. In case of errors the full event payload will be still logged (`dataMessage` just won't be reassigned to a formatted message below), plus, additionally this error massage is logged.